e2e-tests/IIoTPlatform-E2E-Tests/Orchestrated/C_DiscoverEndpointsTestTheory.cs

```diff
@@ -0,0 +1,160 @@
+        private readonly CancellationTokenSource _cancellationToken;
+        private readonly RestClient _restClient;
+        private readonly List<dynamic> _servers;
+
+        public C_DiscoverEndpointsTestTheory(IIoTMultipleNodesTestContext context, ITestOutputHelper output) {
+            _output = output ?? throw new ArgumentNullException(nameof(output));
+            _context = context ?? throw new ArgumentNullException(nameof(context));
+            _context.OutputHelper = _output;
+            _cancellationToken = new CancellationTokenSource(TestConstants.MaxTestTimeoutMilliseconds);
+            _restClient = new RestClient(_context.IIoTPlatformConfigHubConfig.BaseUrl) { Timeout = TestConstants.DefaultTimeoutInMilliseconds };
+
+            // Switch to Orchestrated mode
+            TestHelper.SwitchToOrchestratedModeAsync(_context).GetAwaiter().GetResult();
+
+            // Get OAuth token
+            var token = TestHelper.GetTokenAsync(_context, _cancellationToken.Token).GetAwaiter().GetResult();
+            Assert.NotEmpty(token);
+
+            // Get info about servers
+            var simulatedOpcServers = TestHelper.GetSimulatedPublishedNodesConfigurationAsync(_context, _cancellationToken.Token).GetAwaiter().GetResult();
+            var urls = simulatedOpcServers.Values.ToList().Select(s => s.EndpointUrl).ToList();
+            AddTestOpcServers(urls);
+            dynamic result = TestHelper.WaitForDiscoveryToBeCompletedAsync(_context, _cancellationToken.Token, requestedEndpointUrls: urls).GetAwaiter().GetResult();
+            _servers = result.items;
+
+            // Remove servers
+            var applicationIds = _servers.Select(s => s.applicationId?.ToString());
+            RemoveAllApplications(applicationIds.OfType<string>().ToList());
+        }
+
+        [Fact, PriorityOrder(0)]
+        public void Test_Discover_OPC_UA_Endpoints_IpAddress() {
+            // Add 1 server
+            var server = _servers[0];
+            string url = Convert.ToString(server.discoveryUrls[0]).TrimEnd('/');
+            var urls = new List<string> { url };
+            AddTestOpcServers(urls);
+
+            // Registers servers by running a discovery scan 
+            string ipAddress = Convert.ToString(server.hostAddresses[0]);
+            var cidr = ipAddress.Replace(":50000/", "") + "/16";
+            var body = new {
+                configuration = new {
+                    addressRangesToScan = cidr
+                }
+            };
+            CallRestApi(Method.POST, TestConstants.APIRoutes.RegistryDiscover, body);
+
+            // Validate that the endpoint can be found
+            var result = TestHelper.WaitForEndpointDiscoveryToBeCompleted(_context, _cancellationToken.Token, requestedEndpointUrls: urls).GetAwaiter().GetResult();
+            Assert.Equal(url, ((string)result.items[0].registration.endpointUrl).TrimEnd('/'));
+        }
+
+        [Fact, PriorityOrder(2)]
```

Comment:
Please change the priority to `1`. At the moment you have the following priorities: `0`, `2` and `2`.